### DISCUSSION TITLE
Upsert with multiply onConflict

### DISCUSSION THREAD

**Discussion by Minhnhat0408** [supabase/supabase#18503] [Questions]
So I have a reactions table like this
<img width="989" alt="image" src="https://github.com/supabase/supabase/assets/63915841/e0cda91a-3f01-46c3-9cf6-18a5e9f5e2ac">

And i want to upsert if user click 'up' or  'down' (type)  each reactions can should have at least post_id or user_id different from each other (each user can only have on react on the post)
I come across the solution with upsert  (insert when not exist update when existed) with onConflict ) and a stackoverflow post about multiple constrant for it:
<img width="686" alt="image" src="https://github.com/supabase/supabase/assets/63915841/4e54c9b6-244a-4a19-84c9-1fa47232600d">

but for my case it doesn't work even if I had already set post_id and user_id tobe unique in the table. It still give me the error:"there is no unique or exclusion constraint matching the ON CONFLICT specification"
<img width="435" alt="image" src="https://github.com/supabase/supabase/assets/63915841/243b472a-7432-4691-984c-7834112e34b5">

hope you guys can help me did I do anything wrong

**Comment by GaryAustin1** [comment]
I believe both columns have to be part of a unique constraint, not each separately.
https://www.tutorialsteacher.com/postgresql/unique-constraint

```sql
ALTER TABLE table_name
ADD CONSTRAINT constraint_name
UNIQUE(column1,column2);
```

**Reply by maxhsu-da** [reply]
can this be done in the UI of supabse?

**Reply by GaryAustin1** [reply]
Not the table UI.  The SQL editor allows you to run the code above.

**Reply by Bisqvi** [reply]
If I'm understanding what you're trying to achieve, you can (with the Supabase UI) check these two columns as your primary key, rather than having a separate UUID (if you really mean that there should only be one row in the table for each combo of user_id & post_id). For me, onConflict based on those two columns works then!

**Reply by GaryAustin1** [reply]
Yes, If you want them to both be the primary key then there will be a constraint.
